plugins/default/lib/disk_usage.py

```diff
@@ -1,35 +1,95 @@
 #!/usr/bin/env python
 import re
 import sys
+import json
+import argparse
 from subprocess import CalledProcessError
+from collections import Counter
 
-from openshift import oc
+import openshift
 import nagios
 
 
-check_disk_cmd = ("df", "--output=target,pcent,ipcent")
+check_disk_cmd = ("df", "--output=pcent,ipcent,target")
 # Example output:
 # /etc/hosts      19%    8%
-check_disk_output_pattern = re.compile(r"([\w/]+)\s+(\d+)%\s+(\d+)%")
+check_disk_output_pattern = re.compile(r"\s*(\d+)%\s+(\d+)%\s+(.+)$")
 
 
-def parse_df(line):
+def generate_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-w", "--warn", type=int, action="store",
+                        required=True, help="Warning Threshold")
+    parser.add_argument('-c', '--crit', type=int, action="store",
+                        required=True, help="Critical Threshold")
+    return parser
+
+
+def parse_df_line(line):
     mo = check_disk_output_pattern.match(line)
     if mo is None:
         return ()
-    return (mo.group(1), int(mo.group(2)), int(mo.group(3)))
+    return mo.group(3), int(mo.group(1)), int(mo.group(2))
+
+
+def parse_df_lines(lines):
+    return filter(None, map(parse_df_line, lines.splitlines()))
+
+
+def analize(pod, disks, warning_threshold, critical_threshold):
```

Comment:
There are a few problems here:
1. A critical results shouldn't count as a warning as well, hence why I'm returning a nagios result code per check, something is either UNKNOWN, CRIT, WARN, or OK, in that order, it can't be more than one.
2. the original code didn't even run at all, and in particular the construct it suggested with map/analize didn't handle the fact that multiple rows were returned.  If you look at the changes to the main function, its 4 lines vs. 2 but handles the scenarios that weren't covered.
3. I can change it so analise just does the check part and doesn't take the pod name in but its going to get more confusing how the data is recombined in order to have the info it needs to report.